### DISCUSSION TITLE
ZW0202 Error Source

### DISCUSSION THREAD

**Discussion by cperuffo3** [zwave-js/zwave-js#7491] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
Sporadically when asked to execute multiple commands at once, ZWave JS throws the following error and fails to update specific nodes:
```
[setValue] raised ZWaveError (not handled, code Controller_MessageDropped): Failed to send the command after 1 attempts (ZW0202)
```
The nodes this specific error appears on seems to be sporadic and nonspecific to single nodes

1. Fire multiple simultaneous setValue commands
2. Error occurs almost immediately

Anything else we should know?
This issue has popped up after a full network teardown and rebuild to move from a Gen5+ to a ZST39. This problem is difficult to reliably replicate but seems to be related to when I smash the network (sending a turn off command to 22 light switches on the network). I THINK this is probably a "me" problem (or a my network problem.... but the "1 attempts" makes me also wonder if this isnt a bug

### Software versions

Z-Wave JS UI: 3.18.0

Home Assistant Z-Wave Integration: Driver: 14.3.7 || Server: 1.40.2

### Z-Wave Controller (Stick/Dongle/...)

Zooz ZST39 800 Series LR Zwave Stick

### Device information

**Usually part of the problem:**

Manufacturer: Inovelli
Model name: LZW31-SN
Node ID: [2, 4, 5, 6, 7, 8, 9, 10, 11, 12, 14, 15, 16, 17, 19, 20, 21, 26, 28]

Manufacturer: Inovelli
Model name: VZW31-SN
Node ID: [22, 23, 24, 27]

**Not usually part of the problem:** (because they really arent part of the automations that smash the system)

Manufacturer: Leviton
Model name: ZW4-SF
Node ID: [3]

Manufacturer: Ultrapro
Model name: 39348 / 54890 / 54891 / ZW4008
Node ID: [025]

Manufacturer: Ecolink
Model name: DWZWAVE25
Node ID: [029]

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [x] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-12-13.log](https://github.com/user-attachments/files/18128143/zwavejs_2024-12-13.log)

**Comment by AlCalzone** [comment, marked as answer]
I can't see which firmware version you're on. 700 and 800 series have been plagued by a series of issues where the controller fails to send, sometimes sporadically, sometimes when there has been a bit of incoming traffic before.

Sending a command to 22 nodes that also respond is a good way to trigger these issues.

This typically shows up in the logs like the callback just above the log you quoted:
```
2024-12-13T05:04:09.116Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:                           22
                                    transmit status:                       Fail, took 0 ms
                                    routing attempts:                      0
                                    protocol & route speed:                Unknown (0x00)
                                    routing scheme:                        Idle
                                    ACK RSSI:                              0 dBm
                                    ACK channel no.:                       0
                                    TX channel no.:                        0
                                    TX power:                              0 dBm
                                    measured noise floor:                  0 dBm
                                    ACK TX power by destination:           0 dBm
                                    measured RSSI of ACK from destination: 0 dBm
                                    measured noise floor by destination:   0 dBm
2024-12-13T05:04:09.119Z CNTRLR   [Node 019] [setValue] raised ZWaveError (not handled, code Controller_MessageD
                                  ropped): Failed to send the command after 1 attempts (ZW0202)
```

The most recent fix for this has been in SDK 7.23.0, but unfortunately there is no Zooz firmware built with that yet AFAIK.

---

Side note: your background RSSI also seems very high. For 800 series I'd expect values around -110 dBm.
```
2024-12-13T15:01:29.395Z DRIVER « [RES] [GetBackgroundRSSI]
                                    channel 0: -92 dBm
                                    channel 1: -93 dBm
                                    channel 2: -93 dBm
                                    channel 3: -88 dBm
```
I also see a bunch of repeated electric measurements on a timed schedule, which can cause additional load/noise in your network.

For both I recommend working through https://zwave-js.github.io/node-zwave-js/#/troubleshooting/first-steps - it might just help with your original problem too.

**Reply by P-Verbrugge** [reply]
Zooz is taking a really long time to build a new firmware. :-(

**Comment by jankornblum** [comment]
Same here with aotec gen5+ with latest firmware. Happens only sometimes when switching several devices simultaneously.
